go/plugins/ollama/ollama.go

```diff
@@ -0,0 +1,226 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ollama
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/firebase/genkit/go/ai"
+)
+
+const provider = "ollama"
+
+func defineModel(name string, serverAddress string) {
+	meta := &ai.ModelMetadata{
+		Label: "Ollama - " + name,
+		Supports: ai.ModelCapabilities{
+			Multiturn: true,
```

Comment:
also need to support system prompt, ollama supports it (follow up ok)

Reply:
Sounds good, will follow up on this after hack week.

Reply:
What does supporting system prompt mean exactly? Would that be looking for the first system message in the genkit GenerateRequest.Message array?